tests/e2e/utils/ack_bootstrap/ack_deployment.py

```diff
@@ -0,0 +1,142 @@
+    oidc_url = resp["cluster"]["identity"]["oidc"]["issuer"].split("https://")[1]
+
+    trust_policy = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": f"arn:aws:iam::{account_id}:oidc-provider/{oidc_url}"
+                },
+                "Action": "sts:AssumeRoleWithWebIdentity",
+                "Condition": {
+                    "StringEquals": {
+                        f"{oidc_url}:aud": "sts.amazonaws.com",
+                        f"{oidc_url}:sub": [ 
+                            "system:serviceaccount:ack-system:ack-sagemaker-controller",
+                            "system:serviceaccount:ack-system:ack-applicationautoscaling-controller"
+                        ]
+                    }
+                },
+            }
+        ],
+    }
+    return json.dumps(trust_policy)
+
+
+def create_ack_oidc_role(role_name, cluster_name, region):
+    iam_client = get_iam_client(region=region)
+    acc_id = get_account_id()
+    
+    try:
+        resp = iam_client.create_role(
+            RoleName=role_name, AssumeRolePolicyDocument=profile_trust_policy(cluster_name, region, acc_id)
+        )
+        oidc_role_arn = resp["Role"]["Arn"]
+    except Exception as e:
+        print(e)
+        print("WARNING : This might lead to issues but should be fine if the configs are same.")
+        
+    try:
+        iam_client.attach_role_policy(
+            RoleName=role_name, PolicyArn="arn:aws:iam::aws:policy/AmazonSageMakerFullAccess"
+        )
+    except Exception as e:
+        print(e)
+        print("WARNING : This might lead to issues but should be fine if the configs are same.")
+
+    try:
+        policy_name = "sagemaker_studio_full_access"
+        policy = IAMPolicy(name=policy_name, region=region)
+        policy.create(
+            policy_document=load_json_file(
+                "./resources/ack/iam_ack_oidc_sm_studio_policy.json"
+            )
+        )
+    except Exception as e:
+        print(e)
+        print("WARNING : This might lead to issues but should be fine if the configs are same.")
+
+    try:
+        custom_policy_arn = f"arn:aws:iam::{acc_id}:policy/{policy_name}"
+        iam_client.attach_role_policy(
+            RoleName=role_name, PolicyArn=custom_policy_arn
+        )
+    except Exception as e:
+        print(e)
+        print("WARNING : This might lead to issues but should be fine if the configs are same.")
+
+
+def get_role_arn(role_name, region):
+    iam_client = get_iam_client(region=region)
+    resp = iam_client.get_role(RoleName=role_name)
+    oidc_role_arn = resp["Role"]["Arn"]
+    return oidc_role_arn
+
+def get_account_id():
+    return boto3.client("sts").get_caller_identity().get("Account")
+
+def write_params(oidc_role_arn, region, file_path):
+    file_name = "params.env"
+    with open(f"{file_path}/{file_name}", "w") as file:
```

Comment:
see the configure_env_file https://github.com/awslabs/kubeflow-manifests/blob/main/tests/e2e/utils/load_balancer/setup_load_balancer.py#L113